lib/ansible/utils/collection_loader/_collection_finder.py

```diff
@@ -852,7 +868,14 @@ def is_valid_collection_name(collection_name):
 
         collection_name = to_text(collection_name)
 
-        return bool(re.match(AnsibleCollectionRef.VALID_COLLECTION_NAME_RE, collection_name))
+        if collection_name.count(u'.') != 1:
+            return False
+
+        return all(
+            # NOTE: keywords and identifiers are different in differnt Pythons
+            not iskeyword(ns_or_name) and is_python_identifier(ns_or_name)
```

Comment:
This could use a changelog. We might also want to document this in the developing_collections page or improve error messages to give a better clue about what might be wrong.

ansible-galaxy:
```
ERROR! Invalid collection name 'def.coll', name must be in the format <namespace>.<collection>. 
Please make sure namespace and collection name contains characters from [a-zA-Z0-9_] only.
```
Using a now-invalid collection:
```
[WARNING]: errors were encountered during the plugin load for def.coll.my_module: ['invalid collection name (must be of the form namespace.collection): def.coll']
```

Reply:
Fair enough. But I think that changing the error message should go to a separate PR. The resolver already uses this code in one place anyway...

Reply:
Yeah, but in a very limited way. You could initialize + build + install the tarfile + use it, and it would work. I'm fine with changing the error later though.